communication/http/client/check-http-status-code.yml

```diff
@@ -18,7 +18,7 @@ rule:
       - optional:
         - api: atoi
         - api: wininet.HttpQueryInfo
-      - instruction:
+      - basic block:
```

Comment:
Can you share the sample or verify why this regresses?
- Does it use a different instruction (e.g. `and`)?
- Or is this an issue with the rule matching engine?

Reply:
[capa_test.zip](https://github.com/mandiant/capa-rules/files/9984538/capa_test.zip)
Here is the exe file I was running the tests on -- it's just Windows Notepad. I put it in a folder and zipped it because GitHub didn't want me uploading executable files. Let me know if you need anything else!

Reply:
Great, thanks. Is this the test file for all the changed rules here?

Reply:
Yes, this is the test file for all 3 rules I changed.

Reply:
I'll go through these here. This his was a FP before, see e.g.
```asm
.text:0000000140013C44 BA C8 00 00 00    mov     edx, 0C8h       ; Msg
.text:0000000140013C49 48 FF 15 58 41 01+call    cs:SendMessageW
...
.text:0000000140013D0F 85 C0             test    eax, eax
```

for hit
```yaml
check HTTP status code
namespace  communication/http/client
author     @mr-tz
scope      function
mbc        Communication::HTTP Communication::Read Header [C0002.014]
examples   54ac78733552a62d1d05ea4ba3fc604bb49fe000d7fc948da45335b726e64d75:0x10001a20
function @ 0x14001371C
  and:
    os: windows
    subscope:
      and:
        or:
          mnemonic: test @ 0x140013D0F
        or:
          number: 0xC8 = OK @ 0x140013C44
...
```